pkg/kubelet/eviction/helpers.go

```diff
@@ -444,6 +444,51 @@ func podMemoryUsage(podStats statsapi.PodStats) (v1.ResourceList, error) {
 	}, nil
 }
 
+// localEphemeralVolumeNames returns the set of ephemeral volumes for the pod that are local
+func localEphemeralVolumeNames(pod *v1.Pod) []string {
+	result := []string{}
+	for _, volume := range pod.Spec.Volumes {
+		if volume.GitRepo != nil ||
+			(volume.EmptyDir != nil && volume.EmptyDir.Medium != v1.StorageMediumMemory) ||
+			volume.ConfigMap != nil || volume.DownwardAPI != nil {
+			result = append(result, volume.Name)
+		}
+	}
+	return result
+}
+
+// podLocalEphemeralStorageUsage  aggregates pod local ephemeral storage usage and inode consumption for the specified stats to measure.
+func podLocalEphemeralStorageUsage(podStats statsapi.PodStats, pod *v1.Pod, statsToMeasure []fsStatsType) (v1.ResourceList, error) {
+	disk := resource.Quantity{Format: resource.BinarySI}
```

Comment:
Since this function is very similar to podDiskUsage, I think you can reduce the code by using a common function to calculate container usage.
Or just have one function podLocalStorageUsage (.... func volumeNames) to pass a function for checking the volumenames?

Reply:
ok, fixed